src/domain.rs

```diff
@@ -0,0 +1,31 @@
+/// Configuration for domain checks to be performed on either the `Host`
+/// or the `Origin` header.
```

Comment:
This sounds like there's logic in place to restrict its use to `Host` and `Origin` headers. But that's not accurate. Checking those headers is the "why is this code here" (which is great to state) but there's nothing in here that depends/uses "Host"/"Origin" values. Could be used to check for `Sausage` headers too.